src/file.cr

```diff
@@ -377,6 +378,15 @@ class File < IO::FileDescriptor
     stat.size
   end
 
+  # Truncates the file to the specified length. Requires a write file descriptor
+  def truncate(length)
```

Comment:
mmm I have been playing with this and I think I don't understand well how the argument passed to the system call works. For example, this test passes, when I think it should fail (it should leave an empty file, I'd expect):

```crystal
      File.write(filename, "0123456789")
      File.open(filename, "r+") do |f|
        f.read.should eq("0123456789")
        f.rewind
        f.puts("333")
        f.truncate(0)
      end

      File.read(filename).should eq("333\n")
      File.size(filename).should eq(4)
```

Maybe it's interpreted *relative to the current position*? Apparently not, because if I try to pass a negative argument, it fails:

```
Error truncating file '/Users/sergio/Code/crystal/spec/std/data/temp_write.txt': Invalid argument
```

Sorry, I have never done file manipulation directly with system calls so I'm confused :disappointed: